jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_properties_parser.py

```diff
@@ -91,22 +91,20 @@ def test_clean_line(line, desired_value):
     "line, desired_value",
     (
         ("--help", ["--help"]),
-        ("--abc=0", ["--abc", "0"]),
+        ("--abc=0", ["--abc=0"]),
         ("-J job_name", ["-J", "job_name"]),
         ("-v -J job_name", ["-v", "-J", "job_name"]),
         ("-J job_name -v", ["-J", "job_name", "-v"]),
-        ("-a=0", ["-a", "0"]),
-        ("-a = 0", ["-a", "0"]),
+        ("-a=0", ["-a=0"]),
```

Comment:
the slurm version supported by the charms doesn't accept the equal operator with single character parameters. For example, consider the following command:

```bash
srun -w=juju-7561f2-8,juju-7561f2-11 -p osd-slurmd hostname
```

It raises the following error:

```
srun: error: Unable to allocate resources: Invalid node name specified
```

That said, I wonder how useful is to test something that won't be used actually.

Reply:
These are base checks for `split` from the build-it module `shlex`. As stated in the test's docstring, it just aims to record how the function behaves.

Reply:
isn't it redundant them? I suppose the function is already quite well tested when published with Python. The test is named `test_properties_parser`, so I expect it to test the slurmrestd job properties payload, not a Python function behaviour.

Reply:
I agree it is tested since it is built-in in Python, that is not the intention here. This test is just a way to record or document briefly what I've learned about `shlex.split` on this task, it will be available as a reference together with all other components of the property parser.